app/helpers/message_object_helper.rb

```diff
@@ -10,4 +10,13 @@ def self.pdf_name(message_object)
   def self.base_name(message_object)
     message_object.name.present? ? File.basename(message_object.name, File.extname(message_object.name)) : I18n.t('no_message_object_name')
   end
+
+  def self.unique_name_within_message(message_object, other_file_names, pdf: false)
+    file_name = self.base_name(message_object)
+    matches_count = other_file_names.count { |name| /#{file_name}( \(\d+\))?\.\w*/ =~ name }
+
+    file_name += " (#{matches_count})" if matches_count > 0
+
+    file_name + (pdf ? '.pdf' : File.extname(message_object.name))
+  end
```

Comment:
helper sa zvykne pouzivat na user facing veci vo views, toto sa mi zda, ze je interna vec - hodil by som to do toho concernu